e2e-tests/src/test/scala/pl/touk/nussknacker/BatchDataGenerationSpec.scala

```diff
@@ -0,0 +1,74 @@
+package pl.touk.nussknacker
+
+import io.circe.syntax.EncoderOps
+import io.restassured.RestAssured.`given`
+import io.restassured.module.scala.RestAssuredSupport.AddThenToResponse
+import org.scalatest.freespec.AnyFreeSpecLike
+import org.scalatest.matchers.should.Matchers
+import pl.touk.nussknacker.config.WithE2EInstallationExampleRestAssuredUsersExtensions
+import pl.touk.nussknacker.engine.build.ScenarioBuilder
+import pl.touk.nussknacker.engine.spel.SpelExtension._
+import pl.touk.nussknacker.test.{NuRestAssureExtensions, VeryPatientScalaFutures}
+import pl.touk.nussknacker.ui.process.marshall.CanonicalProcessConverter.toScenarioGraph
+
+class BatchDataGenerationSpec
+    extends AnyFreeSpecLike
+    with DockerBasedInstallationExampleNuEnvironment
+    with Matchers
+    with VeryPatientScalaFutures
+    with NuRestAssureExtensions
+    with WithE2EInstallationExampleRestAssuredUsersExtensions {
+
+  private val simpleBatchTableScenario = ScenarioBuilder
+    .streaming("SumTransactions")
+    .source("sourceId", "table", "Table" -> "'transactions'".spel)
+    .emptySink("end", "dead-end")
+
+  private val designerServiceUrl = "http://localhost:8080"
+
+  "Generate file endpoint should generate records with randomized values for scenario with table source" in {
+    val expectedRecordRegex = {
+      val timestampRegex                    = """\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}\.\d{3}""".r
+      val stringRegex                       = """[a-z\d]{100}""".r
+      val decimalWith15Precision2ScaleRegex = """\d{0,13}\.\d{0,2}""".r
+      val recordRegex =
+        s"""\\{"datetime":"$timestampRegex","client_id":"$stringRegex","amount":$decimalWith15Precision2ScaleRegex}""".r
+      s"""\\{"sourceId":"sourceId","record":$recordRegex}""".r
+    }
+    val numberOfRecordsToGenerate = 1
+
+    given()
+      .applicationState(
+        createBatchScenario(simpleBatchTableScenario.name.value)
+      )
+      .when()
+      .request()
+      .basicAuthAdmin()
+      .jsonBody(toScenarioGraph(simpleBatchTableScenario).asJson.spaces2)
+      .post(
+        s"$designerServiceUrl/api/testInfo/${simpleBatchTableScenario.name.value}/generate/$numberOfRecordsToGenerate"
+      )
+      .Then()
+      .statusCode(200)
+      .matchPlainBodyWithRegex(expectedRecordRegex)
```

Comment:
WDYT about doing it like that:
```scala
      .body(
        matchJsonWithRegexValues(
          s"""
             |{
             |  "sourceId": "sourceId",
             |  "record": {
             |    "datetime":"${regexes.localDateRegex}",
             |    "client_id":"[a-z\\\\d]{100}",
             |    "amount":"${regexes.decimalRegex}"
             |  }
             |}""".stripMargin
        )
      )
```

IMO it's much more readable. 

And yes, regexes are not so readable, but BTH regex is not readable by definition :P But we can create well-named `val`s and use them here. 

Eg.
```scala
trait NuRestAssureMatchers {

  object regexes {
    val localDateRegex = "^\\\\d{4}-\\\\d{2}-\\\\d{2}\\\\s\\\\d{2}:\\\\d{2}:\\\\d{2}\\\\.\\\\d{3}$$"
    val zuluDateRegex = "^\\\\d{4}-\\\\d{2}-\\\\d{2}T\\\\d{2}:\\\\d{2}:\\\\d{2}(?:\\\\.\\\\d{3,6})?Z$$"
    // ujson treats all numbers as a double
    val digitsRegex = "^\\\\d+\\\\.0$$"
    val decimalRegex = "^\\\\d+(\\\\.\\\\d+)?([eE]\\\\d+)?$$"
  }

```

Reply:
First of all, the media type for this kind of response should not be a text by application/x-ndjson. 
I think we should create `MatchNdJsonWithRegexValues` based on `MatchJsonWithRegexValues`. 

Currently, we expect that the body is a JSON (and we parse the body to JSON). And the expected part is a single string which is being read as JSON (with potential regex fields).

In the new `MatchNdJsonWithRegexValues` the body will be split by `\n` first and then each line will be parsed as JSON. The expected value will be a list of strings (not a single string).

In this solution we don't compare strings but JSON objects, so we can format the JSON string (in the `then` part) as we want.

Reply:
Done :+1: Had to adjust the matching logic a bit to enable groups - https://github.com/TouK/nussknacker/pull/6195#discussion_r1670475283